pandas/tests/frame/indexing/test_categorical.py

```diff
@@ -243,23 +243,32 @@ def test_setitem_mask_categorical(self, exp_multi_row):
         # category c is kept in .categories
         tm.assert_frame_equal(df, exp_fancy)
 
-    def test_loc_setitem_categorical_values_partial_column_slice(self):
+    def test_loc_setitem_categorical_values_partial_column_slice(
+        self, using_array_manager
+    ):
         # Assigning a Category to parts of a int/... column uses the values of
         # the Categorical
         df = DataFrame({"a": [1, 1, 1, 1, 1], "b": list("aaaaa")})
         exp = DataFrame({"a": [1, "b", "b", 1, 1], "b": list("aabba")})
-        df.loc[1:2, "a"] = Categorical(["b", "b"], categories=["a", "b"])
-        df.loc[2:3, "b"] = Categorical(["b", "b"], categories=["a", "b"])
-        tm.assert_frame_equal(df, exp)
-
-    def test_loc_setitem_single_row_categorical(self):
+        if using_array_manager:
```

Comment:
why is this desired behavior?

Reply:
Pending outcome of the discussion in https://github.com/pandas-dev/pandas/issues/39584, the "a" column has integer dtype and should preserve that, so string categorical cannot be set